crates/apollo-compiler/src/database/hir_db.rs

```diff
@@ -34,75 +35,172 @@ pub trait HirDatabase: InputDatabase + AstDatabase {
     /// Return all the fragments defined in any file.
     fn all_fragments(&self) -> ByName<FragmentDefinition>;
 
+    /// Return schema definition defined in the compiler.
     fn schema(&self) -> Arc<SchemaDefinition>;
 
+    /// Return all object type definitions defined in the compiler.
     fn object_types(&self) -> ByName<ObjectTypeDefinition>;
 
+    /// Return all scalar type definitions defined in the compiler.
     fn scalars(&self) -> ByName<ScalarTypeDefinition>;
 
+    /// Return all enum type definitions defined in the compiler.
     fn enums(&self) -> ByName<EnumTypeDefinition>;
 
+    /// Return all union type definitions defined in the compiler.
     fn unions(&self) -> ByName<UnionTypeDefinition>;
 
+    /// Return all interface type definitions defined in the compiler.
     fn interfaces(&self) -> ByName<InterfaceTypeDefinition>;
 
+    /// Return all directive definitions defined in the compiler.
     fn directive_definitions(&self) -> ByName<DirectiveDefinition>;
 
+    /// Return all input object type definitions defined in the compiler.
     fn input_objects(&self) -> ByName<InputObjectTypeDefinition>;
 
     // Derived from above queries:
-
+    /// Return an operation definition corresponding to the name and file id.
+    /// Result of this query is not cached internally.
+    #[salsa::transparent]
```

Comment:
@SimonSapin brought up that this is still scanning a Vec internally: https://apollograph.slack.com/archives/C02H6SKKP6Y/p1672929069207239

so we could keep this cached for now?

Reply:
yea sounds good!